federated_aws_cli/login.py

```diff
@@ -235,24 +247,32 @@ def callback(self, code, state, token=None, **kwargs):
                     'AWS STS Call failed when attempting to assume role {} '
                     'with amr {} iss {} and aud {}'.format(
                         self.role_arn, *token_vals))
-                message = (
+                logger.error(
                     'Unable to assume role {}. Please select a different '
                     'role.'.format(self.role_arn))
-                self.role_arn = None
+
+                if len(self.role_map.get("roles", [])) <= 1:
+                    self.exit("Sorry, no valid roles available. Shutting down.")
+                else:
+                    self.role_map.get("roles", []).remove(self.role_arn)
+                    self.role_arn = None
             if self.batch:
                 break
 
-        logger.debug(credentials)
+        logger.debug(self.credentials)
         logger.debug("ID token : {}".format(token["id_token"]))
+        logger.debug("The id for this is: {}".format(id(self)))
```

Comment:
This use of `id` is the python builtin not the `id` variable from earlier I think. This returns the [internal python identity number for the self object I guess?](https://docs.python.org/3/library/functions.html#id). Is this what you want to echo out here?

Reply:
It was yes, I spent a lot of time debugging this because the Flask thing starts up in a different thread/process and so had a completely different login object.  >_<

Reply:
That said, I can probably remove this comment now that I've fixed that bug.

Reply:
Ok got it.